src/main/java/gov/cms/mat/cql_elm_translation/controllers/CqlToolsController.java

```diff
@@ -0,0 +1,39 @@
+package gov.cms.mat.cql_elm_translation.controllers;
+
+import gov.cms.mat.cql_elm_translation.exceptions.CqlFormatException;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.cqframework.cql.tools.formatter.CqlFormatterVisitor;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RestController;
+
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.security.Principal;
+
+@Slf4j
+@RestController
+@RequiredArgsConstructor
+public class CqlToolsController {
+
+  @PutMapping("/cql/format")
+  public ResponseEntity<String> formatCql(@RequestBody String cqlData, Principal principal) {
+    var cqlDataStream = new ByteArrayInputStream(cqlData.getBytes());
+    try {
```

Comment:
Since [ByteArrayInputStream](https://docs.oracle.com/en/java/javase/16/docs/api/java.base/java/io/ByteArrayInputStream.html) implements the `AutoCloseable` interface, it can be used in a try-with-resources block to ensure it's closed after the try/catch block finishes execution. 

```suggestion
try (var cqlDataStream = new ByteArrayInputStream(cqlData.getBytes())) {
```